src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int height;
+    private int base1;
+    private int base2;
+    private String color = getColor();
+
+    public IsoscelesTrapezoid(int height, int base1, int base2, String color) {
+        this.height = height;
+        this.base1 = base1;
+        this.base2 = base2;
+        this.color = color;
+    }
+
+    @Override
+    public double areaCalculator() {
+        return (((double) (base1 * base2) / 2) * height);
+    }
```

Comment:
The method name 'areaCalculator' sounds like a verb. According to the checklist, avoid using verbs for method names. Consider renaming it to 'calculateArea' or similar to reflect its purpose.